python/pyiceberg/avro/reader.py

```diff
@@ -267,10 +267,11 @@ def skip(self, decoder: BinaryDecoder) -> None:
 
 
 class StructReader(Reader):
-    __slots__ = ("field_readers", "create_struct", "struct")
+    __slots__ = ("field_readers", "create_struct", "struct", "create_with_keyword", "_field_reader_functions", "_hash")
```

Comment:
```suggestion
    __slots__ = ("field_readers", "create_struct", "struct", "_create_with_keyword", "_field_reader_functions", "_hash")
```

Reply:
Maybe we want to throw in some tests here?

```python
def test_read_struct() -> None:
    mis = io.BytesIO(b"\x18")
    decoder = StreamingBinaryDecoder(mis)

    struct = StructType(NestedField(1, "id", IntegerType(), required=True))
    reader = StructReader(((0, IntegerReader()),), Record, struct)
    result = reader.read(decoder)
    assert repr(result) == "Record[id=12]"
    assert len(reader.__dict__) == 0
```

This way we make sure that the `__dict__` stays empty, and we take advantage of slots.

Reply:
There is a problem with an arbitrary struct reader that uses Record.  

Record doesn't know what the slot names should be as such it still need to rely on __dict__.  

Rusty

Reply:
Yes, I'm afraid so. This is only the case for the `102 partition` field.